modules/core/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -156,6 +160,27 @@
     // public for tests
     public String blobProviderId;
 
+    /**
+     * Returns the MIME type related to the given {@code contentType}, stripping the charset if present and not a text
+     * content type. For instance, "application/pdf; charset=UTF-8" results in "application/pdf".
```

Comment:
A Content-Type is composed of Mime Type and eventually a Charset, if the method is supposed to return a mime type no need to explain that we remove the charset part, this is the purpose of the method

Reply:
That's a good point, in Nuxeo Blob's `blob.getMimeType` is expected to return the MIME type in all cases. When there's a charset it should be put in the "encoding" part, available with `blob.getEncoding`. 

I didn't understand why the S3 metadata has a `metadata.getContentEncoding()` that's redundant with the `metadata.getContentType()` but actually we use `metadata.getContentEncoding()` wrong, it's something else related to the HTTP transfer itself... So there's another bug lurking here, and we never extract properly the charset(=encoding) from the S3 metadata info

Reply:
We've agreed to:
- not use `metadata.getContentEncoding()`
- split `metadata.getContentType()` into the MIME type and charset, then dispatch them respectively to the blob's MIME type and encoding properties